yt_dlp/extractor/rcti.py

```diff
@@ -253,3 +316,44 @@ def _real_extract(self, url):
                                      display_id, 'Downloading extra entries', metadata))
 
         return self.playlist_result(itertools.chain(*entries), series_id, series_meta.get('title'), series_meta.get('summary'), **metadata)
+
+
+class RCTIPlusTVIE(RCTIPlusBaseIE):
+    _VALID_URL = r'https://www\.rctiplus\.com/((tv/(?P<tvname>\w+))|(?P<eventname>live-event|missed-event))'
+    _TESTS = [{
+        'url': 'https://www.rctiplus.com/tv/rcti',
+        'info_dict': {
+            'id': 'v_lt1',
+            'title': 'RCTI',
+            'ext': 'mp4',
+            'timestamp': 1546344000,
+            'upload_date': '20190101',
+        },
+        'params': {
+            'skip_download': True,
+            'format': 'bestvideo',
+        }
+    }, {
+        # Returned video will always change
+        'url': 'https://www.rctiplus.com/live-event',
+        'only_matching': True,
+    }, {
+        # Returned video will also always change
+        'url': 'https://www.rctiplus.com/missed-event',
+        'only_matching': True,
+    }]
+
+    @classmethod
+    def suitable(cls, url):
+        return False if RCTIPlusIE.suitable(url) else super(RCTIPlusTVIE, cls).suitable(url)
+
+    def _real_extract(self, url):
+        match = re.match(self._VALID_URL, url).groupdict()
+        tv_id = match.get('tvname') or match.get('eventname')
+        webpage = self._download_webpage(url, tv_id)
+        event_re = re.search(
+            r'url ?: ?"https://api\.rctiplus\.com/api/v./(?P<type>[^/]+)/(?P<id>\d+)/url', webpage)
+        if not event_re:
+            raise RegexNotFoundError('Unable to extract video link')
+        video_type, video_id = event_re.groups()
+        return self.url_result('https://www.rctiplus.com/%s/%s/%s' % (video_type, video_id, tv_id), 'RCTIPlus')
```

Comment:
btw, u can use fstrings to simplify this and the api callls if u want

Reply:
done for long api calls, though i don't think i'd want to replace everything

Reply:
done for long api calls, though i don't think i'd want to replace everything

Reply:
sure, just letting you know its an option